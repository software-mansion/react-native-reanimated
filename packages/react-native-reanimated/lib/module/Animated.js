'use strict';

export { ReanimatedFlatList as FlatList } from './component/FlatList';
export { AnimatedImage as Image } from './component/Image';
export { AnimatedScrollView as ScrollView } from './component/ScrollView';
export { AnimatedText as Text } from './component/Text';
export { AnimatedView as View } from './component/View';
export { addWhitelistedNativeProps, addWhitelistedUIProps } from './ConfigHelper';
export { createAnimatedComponent } from './createAnimatedComponent';
//# sourceMappingURL=Animated.js.map