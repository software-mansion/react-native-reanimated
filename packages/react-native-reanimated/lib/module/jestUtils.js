/* eslint-disable @typescript-eslint/no-namespace */
'use strict';

import { IS_JEST, logger, ReanimatedError } from "./common/index.js";
const defaultFramerateConfig = {
  fps: 60
};
const isEmpty = obj => !obj || Object.keys(obj).length === 0;
const getStylesFromObject = obj => {
  return obj === undefined ? {} : Object.fromEntries(Object.entries(obj).map(([property, value]) => [property, value._isReanimatedSharedValue ? value.value : value]));
};
const getCurrentProps = component => {
  const propsObject = component.props.jestAnimatedProps?.value;
  return propsObject ? {
    ...propsObject
  } : {};
};
const getCurrentStyle = component => {
  const styleObject = component.props.style;
  let currentStyle = {};
  if (Array.isArray(styleObject)) {
    // It is possible that style may contain nested arrays. Currently, neither `StyleSheet.flatten` nor `flattenArray` solve this issue.
    // Hence, we're not handling nested arrays at the moment - this is a known limitation of the current implementation.
    styleObject.forEach(style => {
      currentStyle = {
        ...currentStyle,
        ...style
      };
    });
  }
  const jestInlineStyles = component.props.jestInlineStyle;
  const jestAnimatedStyleValue = component.props.jestAnimatedStyle?.value;
  if (Array.isArray(jestInlineStyles)) {
    for (const obj of jestInlineStyles) {
      if ('jestAnimatedValues' in obj) {
        continue;
      }
      const inlineStyles = getStylesFromObject(obj);
      currentStyle = {
        ...currentStyle,
        ...inlineStyles
      };
    }
    currentStyle = {
      ...currentStyle,
      ...jestAnimatedStyleValue
    };
    return currentStyle;
  }
  const inlineStyles = getStylesFromObject(jestInlineStyles);
  currentStyle = isEmpty(jestAnimatedStyleValue) ? {
    ...inlineStyles
  } : {
    ...jestAnimatedStyleValue
  };
  return currentStyle;
};
const checkEqual = (current, expected) => {
  if (Array.isArray(expected)) {
    if (!Array.isArray(current) || expected.length !== current.length) {
      return false;
    }
    for (let i = 0; i < current.length; i++) {
      if (!checkEqual(current[i], expected[i])) {
        return false;
      }
    }
  } else if (typeof current === 'object' && current) {
    if (typeof expected !== 'object' || !expected) {
      return false;
    }
    for (const property in expected) {
      if (!checkEqual(current[property], expected[property])) {
        return false;
      }
    }
  } else {
    return current === expected;
  }
  return true;
};
const findStyleDiff = (current, expected, shouldMatchAllProps) => {
  const diffs = [];
  let isEqual = true;
  let property;
  for (property in expected) {
    if (!checkEqual(current[property], expected[property])) {
      isEqual = false;
      diffs.push({
        property,
        current: current[property],
        expect: expected[property]
      });
    }
  }
  if (shouldMatchAllProps && Object.keys(current).length !== Object.keys(expected).length) {
    isEqual = false;
    // eslint-disable-next-line @typescript-eslint/no-shadow
    let property;
    for (property in current) {
      if (expected[property] === undefined) {
        diffs.push({
          property,
          current: current[property],
          expect: expected[property]
        });
      }
    }
  }
  return {
    isEqual,
    diffs
  };
};
const compareAndFormatDifferences = (currentValues, expectedValues, shouldMatchAllProps = false) => {
  const {
    isEqual,
    diffs
  } = findStyleDiff(currentValues, expectedValues, shouldMatchAllProps);
  if (isEqual) {
    return {
      message: () => 'ok',
      pass: true
    };
  }
  const currentValuesStr = JSON.stringify(currentValues);
  const expectedValuesStr = JSON.stringify(expectedValues);
  const differences = diffs.map(diff => `- '${diff.property}' should be ${JSON.stringify(diff.expect)}, but is ${JSON.stringify(diff.current)}`).join('\n');
  return {
    message: () => `Expected: ${expectedValuesStr}\nReceived: ${currentValuesStr}\n\nDifferences:\n${differences}`,
    pass: false
  };
};
const compareProps = (component, expectedProps) => {
  if (component.props.jestAnimatedProps && Object.keys(component.props.jestAnimatedProps.value).length === 0) {
    return {
      message: () => `Component doesn't have props.`,
      pass: false
    };
  }
  const currentProps = getCurrentProps(component);
  return compareAndFormatDifferences(currentProps, expectedProps);
};
const compareStyle = (component, expectedStyle, config) => {
  if (!component.props.style) {
    return {
      message: () => `Component doesn't have a style.`,
      pass: false
    };
  }
  const {
    shouldMatchAllProps
  } = config;
  const currentStyle = getCurrentStyle(component);
  return compareAndFormatDifferences(currentStyle, expectedStyle, shouldMatchAllProps);
};
let frameTime = Math.round(1000 / defaultFramerateConfig.fps);
const beforeTest = () => {
  jest.useFakeTimers();
};
const afterTest = () => {
  jest.runOnlyPendingTimers();
  jest.useRealTimers();
};
export const withReanimatedTimer = animationTest => {
  logger.warn('This method is deprecated, you should define your own before and after test hooks to enable jest.useFakeTimers(). Check out the documentation for details on testing');
  beforeTest();
  animationTest();
  afterTest();
};
export const advanceAnimationByTime = (time = frameTime) => {
  logger.warn('This method is deprecated, use jest.advanceTimersByTime directly');
  jest.advanceTimersByTime(time);
  jest.runOnlyPendingTimers();
};
export const advanceAnimationByFrame = count => {
  logger.warn('This method is deprecated, use jest.advanceTimersByTime directly');
  jest.advanceTimersByTime(count * frameTime);
  jest.runOnlyPendingTimers();
};
const requireFunction = IS_JEST ? require : () => {
  throw new ReanimatedError('`setUpTests` is available only in Jest environment.');
};
export const setUpTests = (userFramerateConfig = {}) => {
  let expect = global.expect;
  if (expect === undefined) {
    const expectModule = requireFunction('expect');
    expect = expectModule;
    // Starting from Jest 28, "expect" package uses named exports instead of default export.
    // So, requiring "expect" package doesn't give direct access to "expect" function anymore.
    // It gives access to the module object instead.
    // We use this info to detect if the project uses Jest 28 or higher.
    if (typeof expect === 'object') {
      const jestGlobals = requireFunction('@jest/globals');
      expect = jestGlobals.expect;
    }
    if (expect === undefined || expect.extend === undefined) {
      expect = expectModule.default;
    }
  }
  const framerateConfig = {
    ...defaultFramerateConfig,
    ...userFramerateConfig
  };
  frameTime = Math.round(1000 / framerateConfig.fps);
  expect.extend({
    toHaveAnimatedProps(component, expectedProps) {
      return compareProps(component, expectedProps);
    }
  });
  expect.extend({
    toHaveAnimatedStyle(component, expectedStyle, config = {}) {
      return compareStyle(component, expectedStyle, config);
    }
  });
};
export const getAnimatedStyle = component => {
  return getCurrentStyle(
  // This type assertion is needed to get type checking in the following
  // functions since `ReactTestInstance` has its `props` defined as `any`.
  component);
};
//# sourceMappingURL=jestUtils.js.map