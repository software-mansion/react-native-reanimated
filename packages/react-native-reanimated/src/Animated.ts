'use strict';
import type { Extrapolate as _Extrapolate } from './interpolateColor';
import type {
  SharedValue as _SharedValue,
  AnimatedTransform as _AnimatedTransform,
  AnimateStyle as _AnimateStyle,
  StylesOrDefault as _StylesOrDefault,
  EasingFunction as _EasingFunction,
} from './commonTypes';
import type { DerivedValue as _DerivedValue } from './hook/useDerivedValue';
import type {
  TransformStyleTypes as _TransformStyleTypes,
  Adaptable as _Adaptable,
  AdaptTransforms as _AdaptTransforms,
  AnimateProps as _AnimateProps,
} from './helperTypes';

import type { AnimatedScrollViewProps as _AnimatedScrollViewProps } from './component/ScrollView';
import type { FlatListPropsWithLayout as _FlatListPropsWithLayout } from './component/FlatList';

export { createAnimatedComponent } from './createAnimatedComponent';
export { AnimatedText as Text } from './component/Text';
export { AnimatedView as View } from './component/View';
export { AnimatedScrollView as ScrollView } from './component/ScrollView';
export { AnimatedImage as Image } from './component/Image';
export { ReanimatedFlatList as FlatList } from './component/FlatList';
export {
  addWhitelistedNativeProps,
  addWhitelistedUIProps,
} from './ConfigHelper';
/**
 * @deprecated Please import `Extrapolate` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type Extrapolate = typeof _Extrapolate;
/**
 * @deprecated Please import `SharedValue` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */

export type SharedValue<T> = _SharedValue<T>;
/**
 * @deprecated Please import `DerivedValue` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type DerivedValue<T> = _DerivedValue<T>;
/**
 * @deprecated Please import `Adaptable` directly from `react-native-reanimated`
 *   instead of `Animated` namespace.
 */
export type Adaptable<T> = _Adaptable<T>;
/**
 * @deprecated Please import `TransformStyleTypes` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type TransformStyleTypes = _TransformStyleTypes;
/**
 * @deprecated Please import `AdaptTransforms` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type AdaptTransforms<T> = _AdaptTransforms<T>;
/**
 * @deprecated Please import `AnimatedTransform` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type AnimatedTransform = _AnimatedTransform;
/**
 * @deprecated Please import `AnimateStyle` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type AnimateStyle<S> = _AnimateStyle<S>;
/**
 * @deprecated Please import `StylesOrDefault` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type StylesOrDefault<S> = _StylesOrDefault<S>;
/**
 * @deprecated Please import `AnimateProps` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type AnimateProps<P extends object> = _AnimateProps<P>;
/**
 * @deprecated Please import `EasingFunction` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type EasingFunction = _EasingFunction;
/**
 * @deprecated Please import `AnimatedScrollViewProps` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type AnimatedScrollViewProps = _AnimatedScrollViewProps;
/**
 * @deprecated Please import `FlatListPropsWithLayout` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type FlatListPropsWithLayout<T> = _FlatListPropsWithLayout<T>;
