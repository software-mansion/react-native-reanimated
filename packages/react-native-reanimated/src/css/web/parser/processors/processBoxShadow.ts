import { processColor } from 'react-native';
import type { BoxShadowValue } from 'react-native';
import type { ValueProcessor } from '../types';
import { maybeAddSuffix } from '../../utils';

const processBoxShadow: ValueProcessor<
  string | ReadonlyArray<BoxShadowValue>
> = (value) => {
  if (typeof value === 'string') {
    return value;
  }

  return value
    .map(
      ({
        offsetX,
        offsetY,
        color = '',
        blurRadius = '',
        spreadDistance = '',
        inset = '',
      }) =>
        [
          maybeAddSuffix(offsetX, 'px'),
          maybeAddSuffix(offsetY, 'px'),
          maybeAddSuffix(blurRadius, 'px'),
          maybeAddSuffix(spreadDistance, 'px'),
          processColor(color),
          inset === undefined ? '' : inset ? 'inset' : 'outset',
        ].join(' ')
    )
    .join(', ');
};

export default processBoxShadow;
