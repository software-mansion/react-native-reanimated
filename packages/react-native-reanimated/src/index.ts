'use strict';

import './publicGlobals';

import * as Animated from './Animated';

export default Animated;

export type {
  DecayAnimation,
  DelayAnimation,
  RepeatAnimation,
  SequenceAnimation,
  SpringAnimation,
  StyleLayoutAnimation,
  TimingAnimation,
  WithDecayConfig,
  WithSpringConfig,
  WithTimingConfig,
} from './animation';
export {
  cancelAnimation,
  defineAnimation,
  withClamp,
  withDecay,
  withDelay,
  withRepeat,
  withSequence,
  withSpring,
  withTiming,
} from './animation';
export type { ParsedColorArray } from './Colors';
export { convertToRGBA, isColor, processColor } from './Colors';
export type {
  AnimatableValue,
  AnimatableValueObject,
  AnimatedKeyboardInfo,
  AnimatedKeyboardOptions,
  AnimatedSensor,
  AnimatedStyle,
  AnimatedTransform,
  AnimateStyle,
  Animation,
  AnimationCallback,
  AnimationObject,
  BaseLayoutAnimationConfig,
  EasingFunction,
  EntryAnimationsValues,
  EntryExitAnimationFunction,
  ExitAnimationsValues,
  IEntryExitAnimationBuilder,
  ILayoutAnimationBuilder,
  LayoutAnimation,
  LayoutAnimationFunction,
  LayoutAnimationStartFunction,
  LayoutAnimationsValues,
  LayoutAnimationType,
  MeasuredDimensions,
  SensorConfig,
  SharedTransitionAnimationsValues,
  SharedValue,
  StyleProps,
  StylesOrDefault,
  TransformArrayItem,
  Value3D,
  ValueRotation,
} from './commonTypes';
export {
  InterfaceOrientation,
  IOSReferenceFrame,
  isWorkletFunction,
  KeyboardState,
  ReduceMotion,
  SensorType,
  SharedTransitionType,
} from './commonTypes';
export type { FlatListPropsWithLayout } from './component/FlatList';
export { LayoutAnimationConfig } from './component/LayoutAnimationConfig';
export type { PerformanceMonitorProps } from './component/PerformanceMonitor';
export { PerformanceMonitor } from './component/PerformanceMonitor';
export { ReducedMotionConfig } from './component/ReducedMotionConfig';
export type { AnimatedScrollViewProps } from './component/ScrollView';
export { configureReanimatedLogger } from './ConfigHelper';
export type { WorkletRuntime } from './core';
export {
  createWorkletRuntime,
  enableLayoutAnimations,
  executeOnUIRuntimeSync,
  getViewProp,
  isConfigured,
  isReanimated3,
  makeMutable,
  makeShareableCloneRecursive,
  runOnJS,
  runOnRuntime,
  runOnUI,
} from './core';
export type {
  EasingFactoryFn,
  EasingFn,
  EasingFunctionFactory,
} from './Easing';
export { Easing } from './Easing';
export type { FrameInfo } from './frameCallback';
export type {
  Adaptable,
  AdaptTransforms,
  AnimatedProps,
  AnimatedStyleProp,
  AnimateProps,
  TransformStyleTypes,
} from './helperTypes';
export type {
  AnimatedRef,
  DerivedValue,
  EventHandler,
  EventHandlerProcessed,
  FrameCallback,
  GestureHandlers,
  ReanimatedEvent,
  ScrollEvent,
  ScrollHandler,
  ScrollHandlerProcessed,
  ScrollHandlers,
  UseHandlerContext,
} from './hook';
export {
  useAnimatedGestureHandler,
  useAnimatedKeyboard,
  useAnimatedProps,
  useAnimatedReaction,
  useAnimatedRef,
  useAnimatedScrollHandler,
  useAnimatedSensor,
  useAnimatedStyle,
  useComposedEventHandler,
  useDerivedValue,
  useEvent,
  useFrameCallback,
  useHandler,
  useReducedMotion,
  useScrollViewOffset,
  useSharedValue,
  useWorkletCallback,
} from './hook';
export type {
  InterpolateConfig,
  InterpolateHSV,
  InterpolateRGB,
  InterpolationOptions,
} from './interpolateColor';
export {
  ColorSpace,
  /** @deprecated Please use {@link Extrapolation} instead. */
  Extrapolate,
  interpolateColor,
  useInterpolateConfig,
} from './interpolateColor';
export type { ExtrapolationConfig, ExtrapolationType } from './interpolation';
export { clamp, Extrapolation, interpolate } from './interpolation';
export { isSharedValue } from './isSharedValue';
export {
  advanceAnimationByFrame,
  advanceAnimationByTime,
  getAnimatedStyle,
  setUpTests,
  withReanimatedTimer,
} from './jestUtils';
export {
  BaseAnimationBuilder,
  // Bounce
  BounceIn,
  BounceInDown,
  BounceInLeft,
  BounceInRight,
  BounceInUp,
  BounceOut,
  BounceOutDown,
  BounceOutLeft,
  BounceOutRight,
  BounceOutUp,
  combineTransition,
  ComplexAnimationBuilder,
  CurvedTransition,
  EntryExitTransition,
  // Fade
  FadeIn,
  FadeInDown,
  FadeInLeft,
  FadeInRight,
  FadeInUp,
  FadeOut,
  FadeOutDown,
  FadeOutLeft,
  FadeOutRight,
  FadeOutUp,
  FadingTransition,
  FlipInEasyX,
  FlipInEasyY,
  FlipInXDown,
  // Flip
  FlipInXUp,
  FlipInYLeft,
  FlipInYRight,
  FlipOutEasyX,
  FlipOutEasyY,
  FlipOutXDown,
  FlipOutXUp,
  FlipOutYLeft,
  FlipOutYRight,
  JumpingTransition,
  Keyframe,
  // Transitions
  Layout,
  LightSpeedInLeft,
  // Lightspeed
  LightSpeedInRight,
  LightSpeedOutLeft,
  LightSpeedOutRight,
  LinearTransition,
  // Pinwheel
  PinwheelIn,
  PinwheelOut,
  // Roll
  RollInLeft,
  RollInRight,
  RollOutLeft,
  RollOutRight,
  // Rotate
  RotateInDownLeft,
  RotateInDownRight,
  RotateInUpLeft,
  RotateInUpRight,
  RotateOutDownLeft,
  RotateOutDownRight,
  RotateOutUpLeft,
  RotateOutUpRight,
  SequencedTransition,
  // SET
  SharedTransition,
  SlideInDown,
  SlideInLeft,
  // Slide
  SlideInRight,
  SlideInUp,
  SlideOutDown,
  SlideOutLeft,
  SlideOutRight,
  SlideOutUp,
  // Stretch
  StretchInX,
  StretchInY,
  StretchOutX,
  StretchOutY,
  // Zoom
  ZoomIn,
  ZoomInDown,
  ZoomInEasyDown,
  ZoomInEasyUp,
  ZoomInLeft,
  ZoomInRight,
  ZoomInRotate,
  ZoomInUp,
  ZoomOut,
  ZoomOutDown,
  ZoomOutEasyDown,
  ZoomOutEasyUp,
  ZoomOutLeft,
  ZoomOutRight,
  ZoomOutRotate,
  ZoomOutUp,
} from './layoutReanimation';
export { LogLevel as ReanimatedLogLevel } from './logger';
export { startMapper, stopMapper } from './mappers';
export type { ComponentCoords } from './platformFunctions';
export {
  dispatchCommand,
  getRelativeCoords,
  measure,
  scrollTo,
  setGestureState,
  setNativeProps,
} from './platformFunctions';
export { getUseOfValueInStyleWarning } from './pluginUtils';
export { createAnimatedPropAdapter } from './PropAdapters';
export type {
  AnimatedScreenTransition,
  GoBackGesture,
  ScreenTransitionConfig,
} from './screenTransition';
export {
  finishScreenTransition,
  ScreenTransition,
  startScreenTransition,
} from './screenTransition';
