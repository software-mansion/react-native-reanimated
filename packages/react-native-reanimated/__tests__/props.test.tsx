import { View, Pressable, Text } from 'react-native';
import type { ViewStyle } from 'react-native';
import { render, fireEvent } from '@testing-library/react-native';
import Animated, {
  interpolate,
  interpolateColor,
  useSharedValue,
  useAnimatedStyle,
} from '../src';
import { getAnimatedStyle } from '../src/jestUtils';
import { processBoxShadow } from '../src/processBoxShadow';

const AnimatedPressable = Animated.createAnimatedComponent(Pressable);

const AnimatedComponent = () => {
  const pressed = useSharedValue(0);

  const animatedBoxShadow = useAnimatedStyle(() => {
    const blurRadius = interpolate(pressed.value, [0, 1], [10, 0]);
    const color = interpolateColor(
      pressed.value,
      [0, 1],
      ['rgba(255, 0, 0, 1)', 'rgba(0, 0, 255, 1)']
    );

    const boxShadow = `0px 4px ${blurRadius}px 0px ${color}`;

    return {
      boxShadow,
    };
  });

  const handlePress = () => {
    pressed.value = pressed.value === 0 ? 1 : 0;
  };

  return (
    <View
      style={{
        padding: 24,
      }}>
      <AnimatedPressable
        testID={'pressable'}
        style={[
          animatedBoxShadow,
          {
            backgroundColor: 'red',
            padding: 16,
            boxShadow: '0px 4px 10px 0px rgba(255, 0, 0, 1)',
          },
        ]}
        onPress={handlePress}>
        <Text>Button</Text>
      </AnimatedPressable>
    </View>
  );
};

const getDefaultStyle = () => ({
  padding: 16,
  backgroundColor: 'red',
  boxShadow: '0px 4px 10px 0px rgba(255, 0, 0, 1)',
});

const getMultipleBoxShadowStyle = () => ({
  padding: 16,
  backgroundColor: 'red',
  boxShadow:
    '-10px 6px 8px 10px rgba(255, 0, 0, 1), 10px 0px 15px 6px rgba(0, 0, 255, 1)',
});

describe('Test of boxShadow prop', () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.runOnlyPendingTimers();
    jest.useRealTimers();
  });

  test('boxShadow prop animation', () => {
    const style = getDefaultStyle();

    const { getByTestId } = render(<AnimatedComponent />);
    const pressable = getByTestId('pressable');

    expect(pressable.props.style.boxShadow).toBe(
      '0px 4px 10px 0px rgba(255, 0, 0, 1)'
    );
    expect(pressable).toHaveAnimatedStyle(style);
    fireEvent.press(pressable);
    jest.advanceTimersByTime(600);
    style.boxShadow = '0px 4px 0px 0px rgba(0, 0, 255, 1)';
    expect(pressable).toHaveAnimatedStyle(style);
  });

  test('boxShadow prop animation, get animated style', () => {
    const { getByTestId } = render(<AnimatedComponent />);
    const pressable = getByTestId('pressable');

    fireEvent.press(pressable);
    jest.advanceTimersByTime(600);

    const style = getAnimatedStyle(pressable);
    expect((style as ViewStyle).boxShadow).toBe(
      '0px 4px 0px 0px rgba(0, 0, 255, 1)'
    );
  });
  test('one boxShadow string parsing', () => {
    const { getByTestId } = render(<AnimatedComponent />);
    const pressable = getByTestId('pressable');

    expect(pressable.props.style.boxShadow).toBe(
      '0px 4px 10px 0px rgba(255, 0, 0, 1)'
    );

    processBoxShadow(pressable.props.style);

    expect(pressable.props.style.boxShadow).toEqual([
      {
        offsetX: 0,
        offsetY: 4,
        blurRadius: 10,
        spreadDistance: 0,
        color: 'rgba(255, 0, 0, 1)',
      },
    ]);

    const style = getAnimatedStyle(pressable);
    expect((style as ViewStyle).boxShadow).toBe(
      '0px 4px 10px 0px rgba(255, 0, 0, 1)'
    );
  });

  test('two boxShadows string parsing', () => {
    const multipleBoxShadowStyle = getMultipleBoxShadowStyle();

    processBoxShadow(multipleBoxShadowStyle);

    expect(multipleBoxShadowStyle.boxShadow).toEqual([
      {
        offsetX: -10,
        offsetY: 6,
        blurRadius: 8,
        spreadDistance: 10,
        color: 'rgba(255, 0, 0, 1)',
      },
      {
        offsetX: 10,
        offsetY: 0,
        blurRadius: 15,
        spreadDistance: 6,
        color: 'rgba(0, 0, 255, 1)',
      },
    ]);
  });
});
