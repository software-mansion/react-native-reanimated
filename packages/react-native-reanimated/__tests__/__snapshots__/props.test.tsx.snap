// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of boxShadow prop boxShadow prop animation 1`] = `
<View
  style={
    {
      "padding": 24,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    jestAnimatedProps={
      {
        "value": {},
      }
    }
    jestAnimatedStyle={
      {
        "value": {
          "boxShadow": "0px 4px 10px 0px rgba(255, 0, 0, 1)",
        },
      }
    }
    jestInlineStyle={
      [
        {
          "backgroundColor": "red",
          "boxShadow": "0px 4px 10px 0px rgba(255, 0, 0, 1)",
          "padding": 16,
        },
      ]
    }
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "boxShadow": "0px 4px 10px 0px rgba(255, 0, 0, 1)",
        },
        {
          "backgroundColor": "red",
          "boxShadow": "0px 4px 10px 0px rgba(255, 0, 0, 1)",
          "padding": 16,
        },
      ]
    }
    testID="pressable"
  >
    <Text>
      Button
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    jestAnimatedProps={
      {
        "value": {},
      }
    }
    jestAnimatedStyle={
      {
        "value": {
          "boxShadow": "0px 4px 10px rgba(255, 0, 0, 1)",
        },
      }
    }
    jestInlineStyle={
      [
        {
          "backgroundColor": "red",
          "boxShadow": "0px 4px 10px rgba(255, 0, 0, 1)",
          "height": 100,
          "width": 100,
        },
      ]
    }
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "boxShadow": "0px 4px 10px rgba(255, 0, 0, 1)",
        },
        {
          "backgroundColor": "red",
          "boxShadow": "0px 4px 10px rgba(255, 0, 0, 1)",
          "height": 100,
          "width": 100,
        },
      ]
    }
    testID="strippedPressable"
  >
    <Text>
      Stripped
    </Text>
  </View>
</View>
`;

exports[`Test of boxShadow prop boxShadow prop animation, get animated style 1`] = `
<View
  style={
    {
      "padding": 24,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    jestAnimatedProps={
      {
        "value": {},
      }
    }
    jestAnimatedStyle={
      {
        "value": {
          "boxShadow": "0px 4px 10px 0px rgba(255, 0, 0, 1)",
        },
      }
    }
    jestInlineStyle={
      [
        {
          "backgroundColor": "red",
          "boxShadow": "0px 4px 10px 0px rgba(255, 0, 0, 1)",
          "padding": 16,
        },
      ]
    }
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "boxShadow": "0px 4px 10px 0px rgba(255, 0, 0, 1)",
        },
        {
          "backgroundColor": "red",
          "boxShadow": "0px 4px 10px 0px rgba(255, 0, 0, 1)",
          "padding": 16,
        },
      ]
    }
    testID="pressable"
  >
    <Text>
      Button
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    jestAnimatedProps={
      {
        "value": {},
      }
    }
    jestAnimatedStyle={
      {
        "value": {
          "boxShadow": "0px 4px 10px rgba(255, 0, 0, 1)",
        },
      }
    }
    jestInlineStyle={
      [
        {
          "backgroundColor": "red",
          "boxShadow": "0px 4px 10px rgba(255, 0, 0, 1)",
          "height": 100,
          "width": 100,
        },
      ]
    }
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "boxShadow": "0px 4px 10px rgba(255, 0, 0, 1)",
        },
        {
          "backgroundColor": "red",
          "boxShadow": "0px 4px 10px rgba(255, 0, 0, 1)",
          "height": 100,
          "width": 100,
        },
      ]
    }
    testID="strippedPressable"
  >
    <Text>
      Stripped
    </Text>
  </View>
</View>
`;

exports[`Test of boxShadow prop one boxShadow string parsing 1`] = `
<View
  style={
    {
      "padding": 24,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    jestAnimatedProps={
      {
        "value": {},
      }
    }
    jestAnimatedStyle={
      {
        "value": {
          "boxShadow": "0px 4px 10px 0px rgba(255, 0, 0, 1)",
        },
      }
    }
    jestInlineStyle={
      [
        {
          "backgroundColor": "red",
          "boxShadow": "0px 4px 10px 0px rgba(255, 0, 0, 1)",
          "padding": 16,
        },
      ]
    }
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "boxShadow": "0px 4px 10px 0px rgba(255, 0, 0, 1)",
        },
        {
          "backgroundColor": "red",
          "boxShadow": "0px 4px 10px 0px rgba(255, 0, 0, 1)",
          "padding": 16,
        },
      ]
    }
    testID="pressable"
  >
    <Text>
      Button
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    jestAnimatedProps={
      {
        "value": {},
      }
    }
    jestAnimatedStyle={
      {
        "value": {
          "boxShadow": "0px 4px 10px rgba(255, 0, 0, 1)",
        },
      }
    }
    jestInlineStyle={
      [
        {
          "backgroundColor": "red",
          "boxShadow": "0px 4px 10px rgba(255, 0, 0, 1)",
          "height": 100,
          "width": 100,
        },
      ]
    }
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "boxShadow": "0px 4px 10px rgba(255, 0, 0, 1)",
        },
        {
          "backgroundColor": "red",
          "boxShadow": "0px 4px 10px rgba(255, 0, 0, 1)",
          "height": 100,
          "width": 100,
        },
      ]
    }
    testID="strippedPressable"
  >
    <Text>
      Stripped
    </Text>
  </View>
</View>
`;
